### DISCUSSION TITLE
Conditional Logic for User Toggler in WooCommerce Actions Template

### DISCUSSION THREAD

**Discussion by hsankala** [bootscore/bootscore#880] [General]
Yo Basti,

I'm currently upgrading and migrating my WooCommerce site from Bootscore 5.3 to Bootscore 6.0.2. While reviewing the `header.php` file, I noticed that it includes a template part to load the `actions-woocommerce` template part when woocommerce is installed. 

https://github.com/bootscore/bootscore/blob/4e37a62324f616850d53a5998d25a57743cd9d64/header.php#L79-L85

This template part adds a "user toggler" button, which opens a flyout menu on the left-hand side for users to log in.

![image](https://github.com/user-attachments/assets/66fdb82d-e5c9-42f5-8389-e9b40593bc71)

https://github.com/bootscore/bootscore/blob/4e37a62324f616850d53a5998d25a57743cd9d64/template-parts/header/actions-woocommerce.php#L26-L34

However, my site is entirely anonymous — it does not allow login or account creation and only permits anonymous purchasing.

![image](https://github.com/user-attachments/assets/dacc0e8e-92c3-464b-ac8d-c469cd5eac64)

Given this, wouldn't it make sense to update the conditional logic controlling the visibility of this button? Specifically, if the site does not allow account sign-up or login (based on WooCommerce settings), this control shouldn't be shown on the template part page.

Thanks Hefin

**Comment by crftwrk** [comment]
Hi Hefin,

I understand... 

For now, `template-parts/header/actions-woocommerce.php` and `template-parts/header/offcanvas-woocommerce.php` are template files and you can copy them to your child and remove button and offcanvas.

Removing those parts based on WC backend settings seems a bit tricky. But in v.6.1 there is planned to add a lot more filters https://github.com/bootscore/bootscore/pull/841. We can wrap account toggler and offcanvas in a filter, for example ` if (apply_filters('bootscore/enable_account', true))`:


```php
<?php
  if (apply_filters('bootscore/enable_account', true)) {
    if (!is_account_page() && !is_checkout()) { ?>
      <button class="<?= apply_filters('bootscore/class/header/button', 'btn btn-outline-secondary', 'account-toggler'); ?> <?= apply_filters('bootscore/class/header/action/spacer', 'ms-1 ms-md-2', 'account-toggler'); ?> account-toggler" type="button" data-bs-toggle="offcanvas" data-bs-target="#offcanvas-user" aria-controls="offcanvas-user">
        <?= apply_filters('bootscore/icon/user', '<i class="fa-solid fa-user"></i>'); ?> <span class="visually-hidden-focusable">Account</span>
      </button>
    <?php }
  }
?>
```

```php
<?php
if (apply_filters('bootscore/enable_account', true)) {
  if (!is_account_page() && !is_checkout()) { ?>
    <div class="offcanvas offcanvas-<?= apply_filters('bootscore/class/header/offcanvas/direction', 'end', 'account'); ?>" tabindex="-1" id="offcanvas-user">
      <div class="offcanvas-header">
        <span class="h5 offcanvas-title"><?= apply_filters('bootscore/offcanvas/user/title', __('Account', 'bootscore')); ?></span>
        <button type="button" class="btn-close text-reset" data-bs-dismiss="offcanvas" aria-label="Close"></button>
      </div>
      <div class="offcanvas-body">
        <div class="my-offcanvas-account">
          <?= do_shortcode('[woocommerce_my_account]'); ?>
        </div>
      </div>
    </div>
  <?php }
}
?>
```


And then disable account in child's `functions.php`:

```php
/**
 * Disable WooCommerce account
 */
add_filter('bootscore/enable_account', '__return_false');
```

If you agree with that, it would be great if you just fork and submit a pull request to this https://github.com/bootscore/bootscore/tree/Add-more-filters branch.


Would that be a solution?

**Reply by hsankala** [reply]
Yes Basti, that sounds perfect. I agree with creating a filter for this—love the idea of filters across Bootscore! Makes things nicely customizable with simple one-liners in the functions.php file.  I'll create a pull request for this now. Cheers

**Reply by hsankala** [reply]
Awesome pull request submitted. Looking out for it in version 6.1!  Thanks Basti!
https://github.com/bootscore/bootscore/pull/841/commits/93102938514cf1062145a0e731b8e235869ea7b1
